### PR TITLE
tree-sitter-grammars-wasm: init at 0.20.8

### PR DESCRIPTION
**Pull request by aabccd021**
###### Description of changes

[tree-sitter](https://github.com/tree-sitter/tree-sitter) is a code parsing tool and tree-sitter grammars (e.g. ) are the parser grammar for each language.

The tree-sitter grammars can be built into native build (used in native platform such as neovim) or wasm build (mainly used in browser / server-side javascript).
Nix packages for native build is already available on `tree-sitter-grammars.tree-sitter-*`.

This PR adds ones for wasm build on `tree-sitter-grammars-wasm.tree-sitter-*`.

Additional notes:
- [Build instruction for tree-sitter grammar wasm](https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_web/README.md#generate-wasm-language-files)
- LLVM grammar failed to build on my machine, hence omitted
- Some grammar has irregular wasm output file name (e.g `tree-sitter-ql-dbscheme` on PR), so it needs to be overrided manually via `wasmName` variable
- The new file [grammar-wasm.nix](https://github.com/NixOS/nixpkgs/blob/746b9b094f724e1669a2f2121c80b6e2c1cd48c9/pkgs/development/tools/parsing/tree-sitter/grammar-wasm.nix) was made based on similarly behaving [grammar.nix](https://github.com/NixOS/nixpkgs/blob/746b9b094f724e1669a2f2121c80b6e2c1cd48c9/pkgs/development/tools/parsing/tree-sitter/grammar.nix) for easier review

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Can anyone explain what is happening here?

**Comment by aabccd021** [comment]
[tree-sitter](https://github.com/tree-sitter/tree-sitter) is a code parsing tool and tree-sitter grammars (e.g. ) are the parser grammar for each language.

The tree-sitter grammars can be built into native build (used in native platform such as neovim) or wasm build (mainly used in browser / server-side javascript).
Nix packages for native build is already available on `tree-sitter-grammars.tree-sitter-*`.

This PR adds ones for wasm build.

**Comment by aabccd021** [comment]
Additional notes:
- [Build instruction for tree-sitter grammar wasm](https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_web/README.md#generate-wasm-language-files)
- LLVM grammar failed to build on my machine, hence omitted
- Some grammar has irregular wasm output file name (e.g `tree-sitter-ql-dbscheme` on PR), so it needs to be overrided manually via `wasmName` variable
- The new file [grammar-wasm.nix](https://github.com/NixOS/nixpkgs/blob/746b9b094f724e1669a2f2121c80b6e2c1cd48c9/pkgs/development/tools/parsing/tree-sitter/grammar-wasm.nix) was made based on similarly behaving [grammar.nix](https://github.com/NixOS/nixpkgs/blob/746b9b094f724e1669a2f2121c80b6e2c1cd48c9/pkgs/development/tools/parsing/tree-sitter/grammar.nix) for easier review

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -211,6 +211,12 @@
     githubId = 241628;
     name = "Adam Russell";
   };
+  aabccd021 = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:214]: Split this diff in a separate commit.

File: pkgs/development/tools/parsing/tree-sitter/default.nix [hunk 2]
```diff
@@ -47,6 +47,17 @@ let
 
   buildGrammar = callPackage ./grammar.nix { };
 
+  grammarsSrc' = import ./grammars { inherit lib; } // extraGrammars;
```
Comment (AndersonTorres) [inline pkgs/development/tools/parsing/tree-sitter/default.nix:50]: Why these lines were introduced?
They look like adding a functionality not directly related to the title of this pull request. If this is the case, split them into a separate commit, and succinctly explain why they are needed.

File: pkgs/development/tools/parsing/tree-sitter/default.nix [hunk 3]
```diff
@@ -57,18 +68,29 @@ let
           location = grammar.location or null;
           generate = grammar.generate or false;
         };
-      grammars' = import ./grammars { inherit lib; } // extraGrammars;
```
Comment (AndersonTorres) [inline pkgs/development/tools/parsing/tree-sitter/default.nix:60]: Why this was removed?

File: pkgs/development/tools/parsing/tree-sitter/grammar-wasm.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{ stdenv
+, nodejs
+, tree-sitter
+, lib
+, emscripten
+}:
+
```
Comment (AndersonTorres) [inline pkgs/development/tools/parsing/tree-sitter/grammar-wasm.nix:7]:
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by kirillrdy** [comment]
@aabccd021 can you remove `Merge branch 'NixOS:master' into master` commit from this PR and instead rebase on origin/master

**Comment by kirillrdy** [comment]
`The aabccd021:master branch has no history in common with NixOS:master.`

i guess you need to update master in your fork to latest upstream nixpkgs, and rebase again

**Comment by aabccd021** [comment]
@kirillrdy Sorry for the inconvenience. Just did as you said.

**Comment by aabccd021** [comment]
Reopening

**Comment by kirillrdy** [comment]
@aabccd021 no worries, if github wont let you reopen feel free to create a new PR

**Pull request closed without merging**